### ISSUE TITLE
Remove block from Magento 2.2

### ISSUE THREAD

**Issue by juniox** [magento/magento2#11121]
Hi
I have a custom theme which was working fine until 2.1.9. Just upgraded to 2.2 and noticed some blocks I used to remove are now in place.

### Preconditions

1. Ubuntu 14.04
2. Apache 2.4.7
3. PHP 7.0.23
4. MySQL 14.14

### Steps to reproduce

app/design/frontend/MyThemes/custom1/Magento_Theme/layout/default.xml
`<?xml version="1.0"?>`
`<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">`
`  <body>`
    `<referenceBlock name="form.subscribe" remove="true" />`
    `<referenceBlock name="footer_links" remove="true" />`
`  </body>`
`</page>`

### Expected result

Those 2 blocks named "form.subscribe" and "footer_links" should be removed. This same configuration worked before on all 2.1.x versions.

### Actual result

The blocks are now showing on the first page. No modification was done on any theme files, just upgraded from 2.1.9 to 2.2.
Already tried removing both var/ & pub/static content and then deployed static content again, with no success.

Thanks

**Comment by juniox** [comment]
As a workaround I have copied all entries related to removing blocks from my custom Magento_Theme/layout/default.xml to:
`./vendor/magento/theme-frontend-luma/Magento_Theme/layout/default.xml`

It seems my default.xml is somehow not getting triggered.

All these entries worked as expected:
`	<referenceBlock name="form.subscribe" remove="true" />`
`        <referenceBlock name="footer_links" remove="true" />`
`        <referenceBlock name='customer-account-navigation-downloadable-products-link' remove='true'/>`
`        <referenceBlock name="customer-account-navigation-my-credit-cards-link" remove="true"/>`
`        <referenceBlock name="customer-account-navigation-billing-agreements-link" remove="true"/>`
`        <referenceBlock name="customer-account-navigation-newsletter-subscriptions-link" remove="true"/>`
`        <referenceBlock name="header.links" remove="true"/>`

However, this one related to "top.links" did not take any effect. Tried moving it to "footer" instead but it seems there is no such block named "top.links" anymore.

`        <move element="top.links" destination="header.panel"/>`

any idea?
Tks.

**Comment by magento-engcom-team** [comment]
@juniox Thanks for reporting this issue. 
We were not able to reproduce this issue by following the steps you provided. Please provide more detailed steps to reproduce or try to reproduce this issue on a clean installation or latest release.

**Comment by juniox** [comment]
Hello guys,
I´m afraid I dont have further details to provide. Basically I have a custom theme which has been working along all 2.0.x versions. I have just upgrated to 2.2 throughout "Web Setup Wizard" and it completed normally but I noticed my theme was not loading appropriatedly.

After some throubleshooting, I double checked all file/directory permissions, redeploy static content several times, and noticed my app/design/frontend/MyThemes/custom1/Magento_Theme/layout/default.xml was not getting triggered. All other custom css & js was loading as expected but somehow that file default.xml was not triggered.

Since I had a small catalog and could not have my production store offline, I have installed a brand new 2.2 version then copied my theme to this new store and everything worked as expected...

From my perspective we can consider this issue closed, but I dont think this is something normal... nothing else was changed from my theme side...  this issue should not have happened.

Thank you for your support.
